### PR TITLE
[MAINTENANCE] Warnings are errors + Suppress warnings in preparation for supporting SQLAlchemy 2.x

### PR DESCRIPTION
**Pull request by anthonyburdi**
Changes proposed in this pull request:
- Warnings are now errors in our CI pipeline.
- Turn on warnings per the SQLAlchemy 2.0 Support Guide https://docs.sqlalchemy.org/en/20/changelog/migration_20.html#migration-to-2-0-step-two-turn-on-removedin20warnings and suppress warnings. This way we can implement a polish and ratchet approach to step 2 & 3 of the SQLAlchemy 2.0 support guide (https://docs.sqlalchemy.org/en/20/changelog/migration_20.html#x-2-x-migration-overview) and fix each class of warning individually.
- In one case I fixed the test instead of suppressing the warning.

### Definition of Done

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have run any local integration tests and made sure that nothing is broken.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -278,6 +278,8 @@ convention = "google"
 
 [tool.pytest.ini_options]
 filterwarnings = [
+    # Turn all warnings not explicitly filtered below into errors
+    "error",
```
Comment (anthonyburdi) [inline pyproject.toml:282]: This is the line that turns warnings into errors (unless they are specifically filtered out in the lines below).

Comment (anthonyburdi) [reply pyproject.toml:282]: Also, I'm happy to split this into a separate PR if we want to isolate it.

File: tests/integration/usage_statistics/test_usage_stats_common_messages_are_sent_v3api.py [hunk 2]
```diff
@@ -17,6 +17,8 @@
     USAGE_STATISTICS_QA_URL,
 )
 
+yaml: YAMLHandler = YAMLHandler()
```
Comment (Shinnnyshinshin) [inline tests/integration/usage_statistics/test_usage_stats_common_messages_are_sent_v3api.py:20]: 🚀

File: pyproject.toml (hunk 1 above)
Comment (Shinnnyshinshin) [reply pyproject.toml:282]: I think it's within scope of this PR. Happy to keep it as-is.

**Review by Shinnnyshinshin — APPROVED** [review]
LGTM!

**Pull request merged**